twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -185,15 +185,13 @@ def _compile_scala(ctx, label, output, scrooge_jar, deps_java_info,
       expect_java_output = False,
       scalac_jvm_flags = [])
 
-  return java_common.create_provider(
-      use_ijar = False,
-      source_jars = [scrooge_jar],
-      compile_time_jars = depset(
-          [output], transitive = [merged_deps.compile_jars]),
-      transitive_compile_time_jars = depset(
-          [output], transitive = [merged_deps.transitive_compile_time_jars]),
-      transitive_runtime_jars = depset(
-          [output], transitive = [merged_deps.transitive_runtime_jars]))
+  return JavaInfo(
+      source_jar = scrooge_jar,
+      deps = deps_java_info + implicit_deps,
+      runtime_deps = deps_java_info + implicit_deps,
+      exports = deps_java_info + implicit_deps,
```

Comment:
May I ask why do you export everything? Doesn’t sound like the right behavior (though I agree that it looks like the existing code does the same)

Reply:
@johnynek wdyt? As far as I'm concerned we can merge as is but just want to make sure this is indeed what you need

Reply:
I added a comment to the ticket (schemas don't have implementations that users control, so the data types are generally needed by downstream, more-over if any transitive schema changes you will need a recomputation in most cases).

We can revisit if this intuition is wrong.